[build][cgo] pkg-config configs for net-snmp

olivielpeau opened the pull request:
### What does this PR do?

This PR can be regarded as a suggestion on the way we could handle compiling/linking against system vs. embedded libs. Open to comments/criticism, I'm not sure we want to pursue this.

It attempts to make a clear separation between "embedded-libs" builds and "system-libs" builds:
* Adds `pkg-config` configurations for a system install and an embedded install of `net-snmp` (the net-snmp pkgs don't provide one)
* Makes a clear separation between:
  * `embedded` configs that we use when we want to link against the embedded libs
  * `system` configs that are defined for libraries that don't ship `pkg-config` configs when installed system-wide, and that we want to use when linking against the system libs.

For each lib (for now only python and net-snmp), `pkg-config` will use the first config it finds:
* for "embedded" builds, it'll look in the `embedded` dir first, then the `system` dir, then the default dirs of the system
* for "system" builds, it'll look in the `system` dir, then the default dirs of the system

### Motivation

Having hardcoded flags in go source files doesn't work with our model of building binaries against either the embedded libs or the system libs (since it'll link against the first dir where the necessary files are found, so the result is different if an agent is already installed or if it's not).

An issue I encountered was that using the embedded path in the `LDFLAGS` of `snmp.go` also has an impact on how `go build` links against the python interpreter: the hardcoded path to `/opt/datadog-agent6/embedded` in that file creates a binary that's linked against the embedded python interpreter even when `USE_SYSTEM_PY` is used. I'm not sure exactly why to be honest (`cgo` seems to use the flags as expected...).

### Additional Notes

I'm not sure this approach is the best one, but it makes things clearer in terms of:
* what we expect the system to provide
* how we'll link everything

when we build with the system libs, as opposed to when we build with the embedded libs.

So that, if you use `USE_SYSTEM_PY=true` with the build commands, the resulting binary will use the system libraries, and if you don't, the binary will use at runtime what's in `/opt/datadog-agent6/` (and at build-time will expect all the libs to be there).

TODO If this is a path we want to pursue:
* document a bit the differences between `embedded` and `system` configs
* make the `bin/agent/agent` shell script define the correct `PYTHONHOME` depending on the build type
* this has not been tested on Windows, open to any suggestion on how to handle windows, cc @derekwbrown

olivielpeau committed:
[build] pkg-config configs for net-snmp

Clear separation between the system configs and the embedded ones.

masci reviewed (changes requested):
If I got this correctly we should probably rename `USE_SYSTEM_PY`, otherwise this looks very good to me!

olivielpeau committed:
[build] Rename USE_SYSTEM_PY to USE_SYSTEM_LIBS

Makes more sense since python is not the only lib that's affected
by the setting anymore.

truthbk reviewed (approved):
Looks good to me 👍

olivielpeau committed:
[build] Add README in pkg-config dir

[skip ci]
